src/libraries/System.Net.NameResolution/tests/FunctionalTests/GetHostAddressesTest.cs

```diff
@@ -174,19 +174,33 @@ public async Task DnsGetHostAddresses_PreCancelledToken_Throws()
         [OuterLoop]
         [Fact]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/43816")] // Race condition outlined below.
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/33378", TestPlatforms.AnyUnix)] // Cancellation of an outstanding getaddrinfo is not supported on *nix.
         public async Task DnsGetHostAddresses_PostCancelledToken_Throws()
         {
-            using var cts = new CancellationTokenSource();
+            const int numberOfRequests = 100;
+
+            using CancellationTokenSource cts = new();
+            List<Task> tasks = new(capacity: numberOfRequests);
 
-            Task task = Dns.GetHostAddressesAsync(TestSettings.UncachedHost, cts.Token);
+            for (int i = 0; i < numberOfRequests; ++i)
+            {
+                Task task = Dns.GetHostAddressesAsync(TestSettings.UncachedHost, cts.Token);
+                tasks.Add(task);
+            }
 
-            // This test might flake if the cancellation token takes too long to trigger:
-            // It's a race between the DNS server getting back to us and the cancellation processing.
             cts.Cancel();
 
-            OperationCanceledException oce = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => task);
-            Assert.Equal(cts.Token, oce.CancellationToken);
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(async () =>
+            {
+                for (int i = 0; i < numberOfRequests; ++i)
+                {
+                    try
+                    {
+                        await tasks[i].ConfigureAwait(false);
+                    }
+                    catch (Exception ex) when (ex is SocketException)
+                    { }
+                }
```

Comment:
If the first task is canceled, such that the await throws an OperationCanceledException, this construction won't end up waiting on the rest of the tasks, which is something we strive to avoid in both src and tests (leaving incomplete tasks in flight after the associated operation thinks it's done).  It'd be better to do a WhenAll here, and then process the resulting exceptions, e.g.
```C#
try
{
    await Task.WhenAll(tasks);
}
catch
{
    Assert.Contains(tasks, t => t.Exception?.InnerException is OperationCanceledException); 
}
```
or something like that.

Reply:
> leaving incomplete tasks in flight after the associated operation thinks it's done

That's a good point. Thanks.